### PR TITLE
New `pycodestyle.max-line-length` option

### LINKED ISSUE #7644: Add `pydocstyle.line-width` option
Add a new `pydocstyle.line-width` option that overrides the default `line-width` setting. This allows using the formatter with `E501` where `E501` uses another configured line-length. 

Considered alternatives: Creating a new option specific to pycodestyle rather than a formatter-specific option feels more intuitive because the override is specific to this lint rule and then aligns with pydocstyle.max-doc-length.

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR introduces a new `pycodestyl.max-line-length` option that allows overriding the global `line-length` option for `E501` only.

This is useful when using the formatter and `E501` together, where the formatter uses a lower limit and `E501` is only used to catch extra-long lines. 

Closes #7644

## Considerations

~~Our fix infrastructure asserts in some places that the fix doesn't exceed the configured `line-width`. With this change, the question is whether it should use the `pycodestyle.max-line-width` or `line-width` option to make that decision.
I opted for the global `line-width` for now, considering that it should be the lower limit. However, this constraint isn't enforced and users not using the formatter may only specify `pycodestyle.max-line-width` because they're unaware of the global option (and it solves their need).~~

~~I'm interested to hear your thoughts on whether we should use `pycodestyle.max-line-width` or `line-width` to decide on whether to emit a fix or not.~~

Edit: The linter users `pycodestyle.max-line-width`. The `line-width` option has been removed from the `LinterSettings`

## Test Plan

Added integration test. Built the documentation and verified that the links are correct.

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8000** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8000" target="_blank">Graphite</a> 
    * **PR #8002** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8002" target="_blank">Graphite</a> 
      * **PR #8032** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8032" target="_blank">Graphite</a> 
      * **PR #8006** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8006" target="_blank">Graphite</a> 
        * **PR #8035** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8035" target="_blank">Graphite</a> 
          * **PR #8010** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8010" target="_blank">Graphite</a> 
            * **PR #8039** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8039" target="_blank">Graphite</a>  👈
        * **PR #8031** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/8031" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/8039?utm_source=stack-comment).

File: crates/ruff_workspace/src/options.rs
Comment (charliermarsh) [inline crates/ruff_workspace/src/options.rs:2276]: Nit: two spaces between content and `#`

**Review by charliermarsh — APPROVED** [review]
I think any of:

- Global line width
- pycodestyle line width
- Explicit take the min(global line width, pycodestyle line width)

...would all be reasonable choices, so I'm fine to approve with any of those, but if I had to choose, I would say that using the pycodestyle line width is probably the most consistent with the intent, since the _intent_ was to avoid introducing E501 violations.

**Comment by MichaReiser** [comment]
> I would say that using the pycodestyle line width is probably the most consistent with the intent, since the intent was to avoid introducing E501 violations.

Thanks, that's helpful input. It wasn't clear to me what the initial intent was.

File: crates/ruff_workspace/src/options.rs
Comment (MichaReiser) [reply crates/ruff_workspace/src/options.rs:2276]: Doesn't this just apply to Python code?

**Review by charliermarsh — APPROVED** [review]

File: crates/ruff_workspace/src/options.rs
Comment (charliermarsh) [reply crates/ruff_workspace/src/options.rs:2276]: Hmm, I'm not sure what I would do in TOML, actually, but you're probably right.

File: crates/ruff_workspace/src/options.rs
Comment (dhruvmanila) [inline crates/ruff_workspace/src/options.rs:2271]: nit: for consistency with your other PR
```suggestion
    /// The maximum [display width](http://www.unicode.org/reports/tr11/#Overview) to allow for [`line-too-long`](rules/line-too-long.md) violations. By default,
```

**Review by dhruvmanila — APPROVED** [review]

File: crates/ruff_workspace/src/options.rs
Comment (MichaReiser) [reply crates/ruff_workspace/src/options.rs:2278]: I certainly wouldn't do this in TOML. I'm not even used to do it in Python 😆

File: crates/ruff_workspace/src/options.rs
Comment (MichaReiser) [reply crates/ruff_workspace/src/options.rs:2271]: I updated the base PR to use the simpler `width` term. So it should now be consistent. Thanks for flagging.

**Review by charliermarsh — APPROVED** [review]

**Comment by MichaReiser** [comment]
### Merge activity

* **Oct 24, 3:57 AM**: [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/8039) rebased this pull request after merging its parent, because this pull request is set to merge when ready.

**Pull request merged**
